statistics/selectivity.go

```diff
@@ -283,6 +285,53 @@ func (coll *HistColl) Selectivity(ctx sessionctx.Context, exprs []expression.Exp
 			ret *= selectionFactor
 		}
 	}
+
+	// Now we try to cover those still not covered DNF conditions using independence assumption,
+	// i.e., sel(condA or condB) = sel(condA) + sel(condB) - sel(condA) * sel(condB)
+	if mask > 0 {
+		for i := range remainedExprs {
+			if mask&(1<<uint64(i)) == 0 {
+				continue
+			}
+			scalarCond, ok := remainedExprs[i].(*expression.ScalarFunction)
+			// Make sure we only handle DNF condition.
+			if !ok || scalarCond.FuncName.L != ast.LogicOr {
+				continue
+			}
+			dnfItems := expression.FlattenDNFConditions(scalarCond)
+			dnfItems = ranger.MergeDNFItems4Col(ctx, dnfItems)
+
+			selectivity := 0.0
+			for _, cond := range dnfItems {
+				scalar, ok := cond.(*expression.ScalarFunction)
+				if !ok {
+					selectivity = 0.0
+					break
```

Comment:
We can handle `Constant` here, i.e, `a > 1 or 1 or b > 1`. For `Column` / `CorrelatedColumn`, I prefer to ignore this DNF item, i.e, use `continue` instead of `break` since these items are connected with `OR`.

Reply:
Fixed that. Thanks.